include/kokkos-utils/timer/Timer.hpp

```diff
@@ -43,11 +30,46 @@ class Timer
      *       whose device specializations are not @c const.
      */
     template <typename Duration = milliseconds>
-    Duration duration() { return tick.template duration<Duration>(tock); }
+    Duration duration() { return tick->template duration<Duration>(*tock); }
+
+public:
+    //! Start the timer.
+    template <typename U = T>
+    requires std::is_void_v<U>
+    void start()
+    {
+        this->tick = event_t{};
+        this->tick->record();
```

Comment:
Just thinking of `cudaEvent_t` or any "not cheap to create" event type:

```suggestion
        if(! tick.has_value()) tick.emplace();
        tick->record();
```

And as a general remark stop using `this->` everywhere :wink:

Reply:
OK. More efficient indeed!